Assets/Scripts/Bullets/BulletView.cs

```diff
@@ -20,12 +20,12 @@ private void Awake()
 
         private void OnCollisionEnter(Collision collision)
         {
-            if (collision.gameObject.tag == "Enemy")
+            if (collision.gameObject.GetComponent<Interfaces.ITakeDamage>() != null)
             {
-                collision.gameObject.GetComponent<Enemy.EnemyView>().DamageEnemy(bulletController.bulletModel.Damage);
+                collision.gameObject.GetComponent<Interfaces.ITakeDamage>().TakeDamage(bulletController.bulletModel.Damage);
             }
 
-            if (collision.gameObject.tag != "Bullet")
+            if (collision.gameObject.GetComponent<Bullet.BulletView>() == null)
```

Comment:
why do we need this check? we are inside BulletView already ?

Reply:
So 2 bullets dont destroy each other on collision.